api/v1alpha7/conversion.go

```diff
@@ -232,3 +233,31 @@ func (r *OpenStackMachineTemplateList) ConvertFrom(srcRaw ctrlconversion.Hub) er
 	src := srcRaw.(*infrav1.OpenStackMachineTemplateList)
 	return Convert_v1alpha8_OpenStackMachineTemplateList_To_v1alpha7_OpenStackMachineTemplateList(src, r, nil)
 }
+
+func Convert_v1alpha7_OpenStackClusterSpec_To_v1alpha8_OpenStackClusterSpec(in *OpenStackClusterSpec, out *infrav1.OpenStackClusterSpec, s apiconversion.Scope) error {
+	err := autoConvert_v1alpha7_OpenStackClusterSpec_To_v1alpha8_OpenStackClusterSpec(in, out, s)
+	if err != nil {
+		return err
+	}
+
+	if in.ExternalNetworkID != "" {
+		out.ExternalNetwork = infrav1.NetworkFilter{
+			ID: in.ExternalNetworkID,
+		}
+	}
+
+	return nil
+}
+
+func Convert_v1alpha8_OpenStackClusterSpec_To_v1alpha7_OpenStackClusterSpec(in *infrav1.OpenStackClusterSpec, out *OpenStackClusterSpec, s apiconversion.Scope) error {
+	err := autoConvert_v1alpha8_OpenStackClusterSpec_To_v1alpha7_OpenStackClusterSpec(in, out, s)
+	if err != nil {
+		return err
+	}
+
+	if in.ExternalNetwork.ID != "" {
+		out.ExternalNetworkID = in.ExternalNetwork.ID
+	}
+
+	return nil
+}
```

Comment:
Could we have some test coverage of these conversions? I think I'd probably test at the OpenStackCluster level as that's the most relevant unit here, which would also make it easy to expand to cover other conversions.

Reply:
Just to make sure I understand what you suggest: adding unit tests that check that conversion produced an expected OpenStackCluster object. For example like it was done in v1alpha6 for PortOpts convert stuffs?